### PR TITLE
Add `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR adds some documentation on how to test changes to `gha-tools` scripts locally.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Before opening a pull-request, it is recommended to test all changes locally.
+
+The command below is a quick way to set up a test environment for the `gha-tools` repository:
+
+```sh
+docker run \
+  --pull=always \
+  --rm -it \
+  --gpus all \
+  -v /home/user/.aws:/root/.aws:ro \
```
Comment (raydouglass) [inline CONTRIBUTING.md:14]: Maybe use `$HOME/.aws`? Then it can be copied without modification

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  -v /home/user/gha-tools/tools:/root/.local/bin:ro \
+  -v $PWD:/cugraph \
```
Comment (raydouglass) [inline CONTRIBUTING.md:16]: Maybe use `/work` instead of `/cugraph` so it can be copied with modifications

File: tools/rapids-upload-docs
Comment (raydouglass) [inline tools/rapids-upload-docs:1]: Was this file intentionally included in this PR?

Comment (ajschmidt8) [reply tools/rapids-upload-docs:1]: Nope! Removing it.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+The command below is a quick way to set up a disposable test environment for the `gha-tools` repository:
+
+```sh
+docker run \
+  --pull=always \
+  --rm -it \
+  --gpus all \
+  -v $HOME/.aws:/root/.aws:ro \
+  -v /home/user/gha-tools/tools:/root/.local/bin:ro \
```
Comment (raydouglass) [inline CONTRIBUTING.md:15]:
```suggestion
  -v $HOME/gha-tools/tools:/root/.local/bin:ro \
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+Before opening a pull-request, it is recommended to test all changes locally.
+
+The command below is a quick way to set up a disposable test environment for the `gha-tools` repository:
+
+```sh
+docker run \
+  --pull=always \
+  --rm -it \
+  --gpus all \
+  -v $HOME/.aws:/root/.aws:ro \
```
Comment (raydouglass) [inline CONTRIBUTING.md:14]: Need host network for VPN access
```suggestion
  --network=host \
  -v $HOME/.aws:/root/.aws:ro \
  ```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  -w /work \
+  rapidsai/ci:latest
+```
+
+This command makes the following assumptions:
+
+- The `gha-tools` repository is checked out to `/home/user/gha-tools`
+- The current working directory is a RAPIDS repository that has artifacts on [downloads.rapids.ai](https://downloads.rapids.ai) (NVIDIA VPN connectivity is required to access this site), like `cugraph`.
+  - This is important for testing how changes to artifact download scripts will affect local invocations (see note below)
+- The AWS credentials in `/home/user/.aws` have access to the `rapids-downloads` bucket
```
Comment (raydouglass) [inline CONTRIBUTING.md:26]:
```suggestion
- The AWS credentials in `$HOME/.aws` have access to the `rapids-downloads` bucket
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  -v $HOME/.aws:/root/.aws:ro \
+  -v /home/user/gha-tools/tools:/root/.local/bin:ro \
+  -v $PWD:/work \
+  -w /work \
+  rapidsai/ci:latest
+```
+
+This command makes the following assumptions:
+
+- The `gha-tools` repository is checked out to `/home/user/gha-tools`
```
Comment (raydouglass) [inline CONTRIBUTING.md:23]:
```suggestion
- The `gha-tools` repository is checked out to `$HOME/gha-tools`
```

**Comment by ajschmidt8** [comment]
Not sure why the GHAs workflow isn't running :thinking:.

Probably a GH issue. I will admin merge since this is markdown only.

**Pull request merged**